airbyte-integrations/connectors/source-mysql-v2/src/main/kotlin/io/airbyte/integrations/source/mysql/MysqlJdbcEncryption.kt

```diff
@@ -0,0 +1,147 @@
+    }
+
+    private fun prepareClientCertificateKeyStore(): Pair<URI, String>? {
+        var clientCertKeyStorePair: Pair<URI, String>? = null
+
+        if (!clientCertificate.isNullOrEmpty() && !clientKey.isNullOrEmpty()) {
+            val clientKeyPassword = getOrGeneratePassword()
+            try {
+                val clientCertKeyStoreUri =
+                    SSLCertificateUtils.keyStoreFromClientCertificate(
+                        clientCertificate,
+                        clientKey,
+                        clientKeyPassword,
+                        null
+                    )
+                clientCertKeyStorePair = Pair(clientCertKeyStoreUri, clientKeyPassword)
+            } catch (ex: Exception) {
+                throw RuntimeException("Failed to create keystore for Client certificate", ex)
+            }
+        }
+        return clientCertKeyStorePair
+    }
+
+    fun parseSSLConfig(): Map<String, String> {
+        var caCertKeyStorePair: Pair<URI, String>?
+        var clientCertKeyStorePair: Pair<URI, String>?
+        val additionalParameters: MutableMap<String, String> = mutableMapOf()
+
+        additionalParameters[SSL_MODE] = sslMode.jdbcPropertyName
+
+        caCertKeyStorePair = prepareCACertificateKeyStore()
+
+        if (null != caCertKeyStorePair) {
+            log.debug { "uri for ca cert keystore: ${caCertKeyStorePair.first}" }
+            try {
+                additionalParameters.putAll(
+                    mapOf(
+                        TRUST_KEY_STORE_URL to caCertKeyStorePair.first.toURL().toString(),
+                        TRUST_KEY_STORE_PASS to caCertKeyStorePair.second,
+                        TRUST_KEY_STORE_TYPE to KEY_STORE_TYPE_PKCS12
+                    )
+                )
+            } catch (e: MalformedURLException) {
+                throw ConfigErrorException("Unable to get a URL for trust key store")
+            }
+
+            clientCertKeyStorePair = prepareClientCertificateKeyStore()
+
+            if (null != clientCertKeyStorePair) {
+                log.debug {
+                    "uri for client cert keystore: ${clientCertKeyStorePair.first} / ${clientCertKeyStorePair.second}"
+                }
+                try {
+                    additionalParameters.putAll(
+                        mapOf(
+                            CLIENT_KEY_STORE_URL to clientCertKeyStorePair.first.toURL().toString(),
+                            CLIENT_KEY_STORE_PASS to clientCertKeyStorePair.second,
+                            CLIENT_KEY_STORE_TYPE to KEY_STORE_TYPE_PKCS12
+                        )
+                    )
+                } catch (e: MalformedURLException) {
+                    throw ConfigErrorException("Unable to get a URL for client key store")
+                }
+            }
+        }
+        return additionalParameters
+    }
+}
+
+enum class SSLMode(val jdbcPropertyName: String) {
```

Comment:
this `jdbcPropertyName` doesn't seem necessary, it's just the enum name, lowercased;
the body of `fromJdbcPropertyName` can be adjusted to do an `equalsIgnoreCase` instead

Reply:
cleaned it up.